### PR TITLE
:bug: fix issue to scaffold same Kind for different groups with multi-group support by using group name in crd/patches and rbac editor/viewer manifests

### LINKED ISSUE #2495: Kubebuilder with multi-group and same Kind names doesn't generate proper crd/patch and rbac manifests
### What broke? What's expected?

In a multi-group project when 2 groups have the same CRD name for example

```
---
apiVersion: foo.rmb938.com/v1alpha1
kind: Bar
...
---
apiVersion: faz.rmb938.com/v1alpha1
kind: Bar
```

The crd patches generated in `config/crd/patches` and the rbac generated (for viewer and editor) in `config/rbac` are only for the first group and kind that gets generated. They should be generated for all groups.

### Reproducing this issue

Run the following

```
kubebuilder create api --controller=false --group foo --version v1alpha1 --kind Bar
kubebuilder create api --controller=false --group faz --version v1alpha1 --kind Bar
```

### KubeBuilder (CLI) Version

Version: main.version{KubeBuilderVersion:"3.3.0", KubernetesVendor:"1.23.1", GitCommit:"47859bf2ebf96a64db69a2f7074ffdec7f15c1ec", BuildDate:"2022-01-18T17:03:29Z", GoOs:"darwin", GoArch:"amd64"}

### PROJECT version

3

### Plugin versions

```yaml
layout:
- go.kubebuilder.io/v3
```

### Other versions

```
 %  go version
go version go1.17.3 darwin/amd64
```

```
sigs.k8s.io/controller-runtime v0.11.0
```

```
sigs.k8s.io/controller-tools/cmd/controller-gen@v0.8.0
```

```
%  kubectl version
Client Version: version.Info{Major:"1", Minor:"19", GitVersion:"v1.19.2", GitCommit:"f5743093fd1c663cb0cbc89748f730662345d44d", GitTreeState:"clean", BuildDate:"2020-09-16T21:51:21Z", GoVersion:"go1.15.2", Compiler:"gc", Platform:"darwin/amd64"}
Server Version: version.Info{Major:"1", Minor:"21", GitVersion:"v1.21.5-gke.1302", GitCommit:"639f3a74abf258418493e9b75f2f98a08da29733", GitTreeState:"clean", BuildDate:"2021-10-21T21:35:48Z", GoVersion:"go1.16.7b7", Compiler:"gc", Platform:"linux/amd64"}
```

```
%  kustomize version
{Version:kustomize/v4.4.0 GitCommit:63ec6bdb3d737a7c66901828c5743656c49b60e1 BuildDate:2021-09-27T16:13:36Z GoOs:darwin GoArch:amd64}
```

### Extra Labels

_No response_

### PR DESCRIPTION
**Pull request by rmb938**
Modification to include the group name in the crd patch and rbac editor/viewer file names. This is to prevent file naming conflicts when multiple groups have the same kind names.

This is to fix #2495

### CODE REVIEW

**Comment by rmb938** [comment]
I've split the commits into actual code changes vs test data being re-generated. Hopefully it makes it easier to review :)

File: pkg/plugins/common/kustomize/v1/scaffolds/internal/templates/config/crd/patches/enablecainjection_patch.go [hunk 1]
```diff
@@ -33,7 +33,7 @@ type EnableCAInjectionPatch struct {
 // SetTemplateDefaults implements file.Template
 func (f *EnableCAInjectionPatch) SetTemplateDefaults() error {
 	if f.Path == "" {
-		f.Path = filepath.Join("config", "crd", "patches", "cainjection_in_%[plural].yaml")
+		f.Path = filepath.Join("config", "crd", "patches", "cainjection_in_%[group]_%[plural].yaml")
```
Comment (camilamacedo86) [inline pkg/plugins/common/kustomize/v1/scaffolds/internal/templates/config/crd/patches/enablecainjection_patch.go:36]:
```suggestion
		if f.MultiGroup {
			f.Path = filepath.Join("config", "crd", "patches", "cainjection_in_%[group]_%[plural].yaml")
		} else {
			f.Path = filepath.Join("config", "crd", "patches", "cainjection_in_%[plural].yaml")
		}
```

File: pkg/plugins/common/kustomize/v1/scaffolds/internal/templates/config/crd/patches/enablewebhook_patch.go [hunk 2]
```diff
@@ -33,7 +33,7 @@ type EnableWebhookPatch struct {
 // SetTemplateDefaults implements file.Template
 func (f *EnableWebhookPatch) SetTemplateDefaults() error {
 	if f.Path == "" {
-		f.Path = filepath.Join("config", "crd", "patches", "webhook_in_%[plural].yaml")
+		f.Path = filepath.Join("config", "crd", "patches", "webhook_in_%[group]_%[plural].yaml")
```
Comment (camilamacedo86) [inline pkg/plugins/common/kustomize/v1/scaffolds/internal/templates/config/crd/patches/enablewebhook_patch.go:36]:
```suggestion
                if f.MultiGroup {
			f.Path = filepath.Join("config", "crd", "patches", "webhook_in_%[group]_%[plural].yaml")
		} else {
			f.Path = filepath.Join("config", "crd", "patches", "webhook_in_%[plural].yaml")
		}

File: pkg/plugins/common/kustomize/v1/scaffolds/internal/templates/config/rbac/crd_editor_role.go [hunk 3]
```diff
@@ -33,7 +33,7 @@ type CRDEditorRole struct {
 // SetTemplateDefaults implements file.Template
 func (f *CRDEditorRole) SetTemplateDefaults() error {
 	if f.Path == "" {
-		f.Path = filepath.Join("config", "rbac", "%[kind]_editor_role.yaml")
+		f.Path = filepath.Join("config", "rbac", "%[group]_%[kind]_editor_role.yaml")
```
Comment (camilamacedo86) [inline pkg/plugins/common/kustomize/v1/scaffolds/internal/templates/config/rbac/crd_editor_role.go:36]:
```suggestion
                 if f.MultiGroup {
			f.Path = filepath.Join("config", "rbac", "%[group]_%[kind]_editor_role.yaml")
		} else {
			f.Path = filepath.Join("config", "rbac", "%[kind]_editor_role.yaml")
		}
		
```

File: pkg/plugins/common/kustomize/v1/scaffolds/internal/templates/config/rbac/crd_viewer_role.go [hunk 4]
```diff
@@ -33,7 +33,7 @@ type CRDViewerRole struct {
 // SetTemplateDefaults implements file.Template
 func (f *CRDViewerRole) SetTemplateDefaults() error {
 	if f.Path == "" {
-		f.Path = filepath.Join("config", "rbac", "%[kind]_viewer_role.yaml")
+		f.Path = filepath.Join("config", "rbac", "%[group]_%[kind]_viewer_role.yaml")
```
Comment (camilamacedo86) [inline pkg/plugins/common/kustomize/v1/scaffolds/internal/templates/config/rbac/crd_viewer_role.go:36]:
```suggestion
                 if f.MultiGroup {
			f.Path = filepath.Join("config", "rbac", "%[group]_%[kind]_viewer_role.yaml")
		} else {
			f.Path = filepath.Join("config", "rbac", "%[kind]_viewer_role.yaml")
		}
		
```

**Review by camilamacedo86 — CHANGES_REQUESTED** [review]
Thank you for the terrific collab 🥇 
Just a nit. We need to only change the path/name when is multigroup. 
Could you please:
- Apply the suggestions
- Run make generate
- Push the changes in 1 commit ( squashed )

For we are able to get these ones merged
Thank you a lot.

**Comment by rmb938** [comment]
ok @camilamacedo86 All the changes are done and tests have passed. Let me know if there's anything else!

**Review by camilamacedo86 — APPROVED** [review]
Thank you for the great contribution 🥇 

/lgtm
/approved

**Pull request merged**